### PR TITLE
:sparkles: (dmk) [DSDK-542]: Add API calls and models for secure channel

### PR DESCRIPTION
**Pull request by jdabbech-ledger**
### 📝 Description

Manager api calls for secure channel 

### ❓ Context

- **JIRA or GitHub link**: [DSDK-542]

- **Feature**:

### ✅ Checklist

Pull Requests must pass CI checks and undergo code review. Set the PR as Draft if it is not yet ready for review.

- [x] **Covered by automatic tests** 
- [x] **Changeset is provided** 
- [ ] **Impact of the changes:** 
  - _list of the changes_

---

### 🧐 Checklist for the PR Reviewers

- [ ] **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- [ ] **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- [ ] **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- [ ] **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- [ ] **Any new dependencies** have been justified and documented.

[DSDK-542]: https://ledgerhq.atlassian.net/browse/DSDK-542?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/device-management-kit/src/internal/manager-api/model/Firmware.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
+type BaseFirmware = {
+  id: number;
+  name: string;
+  description: string | null | undefined;
```
[the remark this answers is not in the document]
Comment (jdabbech-ledger) [reply packages/device-management-kit/src/internal/manager-api/model/Firmware.ts:4]: As it's a model that comes from API, I guess yes.

File: packages/device-management-kit/src/api/command/os/GetOsVersionCommand.ts [hunk 2]
```diff
@@ -91,7 +91,10 @@ export class GetOsVersionCommand implements Command<GetOsVersionResponse> {
     }
     const parser = new ApduParser(apduResponse);
 
-    const targetId = parser.encodeToHexaString(parser.extractFieldByLength(4));
+    const targetId = parseInt(
```
[the remark this answers is not in the document]
Comment (jdabbech-ledger) [reply packages/device-management-kit/src/api/command/os/GetOsVersionCommand.ts:94]: To convert the hexastring to a decimal string
 I could add a parser method for this

File: packages/device-management-kit/src/internal/manager-api/model/Firmware.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (aussedatlo) [reply packages/device-management-kit/src/internal/manager-api/model/Firmware.ts:4]: Is there any API doc for this endpoint ?

[the remark this answers is not in the document]
Comment (jdabbech-ledger) [reply packages/device-management-kit/src/internal/manager-api/model/Firmware.ts:4]: Not that I know of, this type is the same in LL

[the remark this answers is not in the document]
Comment (aussedatlo) [reply packages/device-management-kit/src/internal/manager-api/model/Firmware.ts:4]: Yeah I was thinking if the null is really useful here or needed for the business logic on the LL side

File: packages/device-management-kit/src/api/command/os/GetOsVersionCommand.test.ts [hunk 3]
```diff
@@ -5,6 +5,7 @@ import {
 import { DeviceModelId } from "@api/device/DeviceModel";
 import { ApduResponse } from "@api/device-session/ApduResponse";
 
+import { getOsVersionCommandResponseStubBuilder } from "./__mocks__/GetOsVersionCommand";
```
Comment (valpinkman) [inline packages/device-management-kit/src/api/command/os/GetOsVersionCommand.test.ts:8]:
```suggestion
import { getOsVersionCommandResponseMockBuilder } from "./__mocks__/GetOsVersionCommand";
```

File: packages/device-management-kit/src/api/command/os/GetOsVersionCommand.test.ts [hunk 4]
```diff
@@ -62,16 +63,7 @@ describe("GetOsVersionCommand", () => {
...
-            targetId: "33000004",
-            seVersion: "2.2.3",
-            seFlags: 3858759680,
-            mcuSephVersion: "2.30",
-            mcuBootloaderVersion: "1.16",
-            hwVersion: "00",
-            langId: "00",
-            recoverState: "00",
-          },
+          data: getOsVersionCommandResponseStubBuilder(DeviceModelId.NANO_X),
```
Comment (valpinkman) [inline packages/device-management-kit/src/api/command/os/GetOsVersionCommand.test.ts:66]:
```suggestion
          data: getOsVersionCommandResponseMockBuilder(DeviceModelId.NANO_X),
```

File: packages/device-management-kit/src/api/command/os/__mocks__/GetOsVersionCommand.ts [hunk 5]
```diff
@@ -0,0 +1,63 @@
+import { DeviceModelId } from "@api/device/DeviceModel";
+import { type GetOsVersionResponse } from "@api/index";
+
+export const getOsVersionCommandResponseStubBuilder = (
```
Comment (valpinkman) [inline packages/device-management-kit/src/api/command/os/__mocks__/GetOsVersionCommand.ts:4]:
```suggestion
export const getOsVersionCommandResponseMockBuilder = (
```

File: packages/device-management-kit/src/internal/manager-api/data/__mocks__/GetDeviceVersion.ts
Comment (valpinkman) [inline packages/device-management-kit/src/internal/manager-api/data/__mocks__/GetDeviceVersion.ts:1]: [SHOULD] Either you mock GetDeviceVersion (we should be placed wherever the real GetDeviceVersion is) or you need to change the name of this file

File: packages/device-management-kit/src/internal/manager-api/data/__mocks__/GetFirmwareVersion.ts
Comment (valpinkman) [inline packages/device-management-kit/src/internal/manager-api/data/__mocks__/GetFirmwareVersion.ts:1]: [SHOULD] same here

File: packages/device-management-kit/src/internal/manager-api/service/DefaultManagerApiService.test.ts [hunk 6]
```diff
@@ -1,5 +1,7 @@
 import { Left, Right } from "purify-ts";
 
+import { getOsVersionCommandResponseStubBuilder } from "@api/command/os/__mocks__/GetOsVersionCommand";
```
Comment (valpinkman) [inline packages/device-management-kit/src/internal/manager-api/service/DefaultManagerApiService.test.ts:3]:
```suggestion
import { getOsVersionCommandResponseMockBuilder } from "@api/command/os/__mocks__/GetOsVersionCommand";
```

**Review by valpinkman — CHANGES_REQUESTED** [review]

File: packages/device-management-kit/src/internal/manager-api/model/Firmware.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (jdabbech-ledger) [reply packages/device-management-kit/src/internal/manager-api/model/Firmware.ts:4]: IMO it's a bit over-engineered to do so, as the dmk doesn't manage the device firmware.
Those manager api calls aim to setup a secure channel by using only the fields `BaseFirmware.perso` and `GetDeviceVersion.id`.
Deleting useless fields is probably clearer than declaring and mapping them for nothing, wdyt?

[the remark this answers is not in the document]
Comment (ofreyssinet-ledger) [reply packages/device-management-kit/src/internal/manager-api/model/Firmware.ts:4]: My 2 cents let's keep the model that comes from the API, then when we need more than the two fields (when the DMK will handle firmware updates) we will do an internal mapping.

**Review by ofreyssinet-ledger — APPROVED** [review]

File: packages/device-management-kit/src/internal/manager-api/data/__mocks__/GetDeviceVersion.ts
Comment (jiyuzhuang) [reply packages/device-management-kit/src/internal/manager-api/data/__mocks__/GetDeviceVersion.ts:1]: I'm not sure whether I understand, you mean renaming like `GetDeviceVersionMock.ts` ?

**Review by valpinkman — APPROVED** [review]

**Pull request merged**